.github/workflows/upstream.yml

```diff
@@ -7,108 +6,137 @@ on:
 
 concurrency:
   group: upstream-${{ github.ref }}
-  cancel-in-progress: true
+  cancel-in-progress: false
 
 env:
-  PIP_ONLY_BINARY: numpy
-  # For cmake:
-  VERBOSE: 1
+  PYTEST_TIMEOUT: 300
 
 jobs:
   standard:
-    name: "🐍 3.11 latest internals • ubuntu-latest • x64"
+    name: "🐍 3.12 latest • ubuntu-latest • x64"
     runs-on: ubuntu-latest
+
+    # Hint: Select the 'python dev' label in the PR web view.
     if: "contains(github.event.pull_request.labels.*.name, 'python dev')"
 
     steps:
+    - name: Show env
+      run: env
+
     - uses: actions/checkout@v3
 
-    - name: Setup Python 3.11
+    - name: Setup Python 3.12
       uses: actions/setup-python@v4
       with:
-        python-version: "3.11-dev"
+        python-version: "3.12-dev"
 
-    - name: Setup Boost (Linux)
-      if: runner.os == 'Linux'
+    - name: Setup Boost
       run: sudo apt-get install libboost-dev
 
     - name: Update CMake
       uses: jwlawson/actions-setup-cmake@v1.13
 
-    - name: Prepare env
+    - name: Run pip installs
       run: |
-        python -m pip install -r tests/requirements.txt
+        python -m pip install --upgrade pip
+        python -m pip install --prefer-binary -r tests/requirements.txt
+    #   python -m pip install --prefer-binary numpy  # SLOW
+    #   python -m pip install --prefer-binary scipy  # FAILED ~Nov 2022
```

Comment:
These generally will only work after RC1 is out, and by that time, we should be including the next release in our regular, required CI. I'd say just keep them out of this.